index.js

```diff
@@ -0,0 +1,166 @@
+// import all necessary modules form node or locally
+const chalk = require('chalk');
+const readline = require('readline-sync');
+const randomcolor = require('randomcolor');
+
+// Object with accepted inputs
+const allowedInput = {
+  color: [
+    'red',
+    'orange',
+    'yellow',
+    'green',
+    'blue',
+    'purple',
+    'pink',
+    'monochrome',
+  ],
+  hue: ['light', 'dark'],
+};
+
+//initialize a random hex color variable and print it out
+let colorChoice = randomcolor();
+const symbol = '#';
+
+// get the user input from the process.argv if supplied//const Args = process.argv[2].split('x');
+// creates an array from third argument onwards
+const Args = process.argv.slice(2);
+
+//initialize height and width
+let measures;
+const minHeight = 5;
+const minWidth = 19;
+let height;
+let width;
+
+//initialize color choices
+let hueChoice = '';
+let luminosityChoice = '';
+
+if (/\d*x\d*/.test(Args[0]) === true) {
+  // test if first argument matches Regex (digtsxdigits)
+  measures = process.argv[2].split('x'); //generate array from string
+  height = parseInt(measures[0]);
+  width = parseInt(measures[1]);
+  if (height < minHeight) {
+    height = minHeight;
+    console.log(`Too small, I set the height to ${minHeight}.`);
+  }
+  if (width < minWidth) {
+    width = minWidth;
+    console.log(`Too small, I set the height to ${minWidth}.`);
+  }
+  hueChoice = Args[1]; //saves hue choice, first user supplied argument
+  luminosityChoice = Args[2];
+} // saves luminosity choice, second user supplied argument
+
+//if the first argument is ask, ask for input:
+else if (Args[0] === 'ask') {
```

Comment:
Style: most JavaScript code has the `else` or `else if` on the same line as the closing bracket:

```
} else if (Args[0] === 'ask') {
```